modules/wallet/addresses.go

```diff
@@ -16,8 +16,13 @@ func (w *Wallet) timelockedCoinAddress(unlockHeight consensus.BlockHeight) (coin
 	spendConditions = consensus.SpendConditions{
 		TimeLock:      unlockHeight,
 		NumSignatures: 1,
-		PublicKeys:    []crypto.PublicKey{pk},
+		PublicKeys: []consensus.SiaPublicKey{
+			consensus.SiaPublicKey{
+				Algorithm: consensus.ED25519Identifier,
+			},
+		},
 	}
+	copy(spendConditions.PublicKeys[0].Key, pk[:])
```

Comment:
That would be the next step.  It doesn't really have much sense to have 'SignatureKeys' function if you're planning on supporting multiple signature types. Instead you'd have a GenerateEd25519Keys(). But even that couldn't return a SiaPublicKey, because you'd have an import cycle between consensus and crypto.

So you'd need a consensus.GenerateSignatureKeys(consensus.IdentifierEd25519)